pkg/apis/fleet.cattle.io/v1alpha1/bundle_types.go

```diff
@@ -221,6 +221,12 @@ type BundleTarget struct {
 	ClusterGroupSelector *metav1.LabelSelector `json:"clusterGroupSelector,omitempty"`
 	// DoNotDeploy if set to true, will not deploy to this target.
 	DoNotDeploy bool `json:"doNotDeploy,omitempty"`
+	// NamespaceLabels are labels that will be appended to the namespace created by Fleet.
+	// +nullable
+	NamespaceLabels *map[string]string `json:"namespaceLabels,omitempty"`
```

Comment:
I see this is already a pointer to a map in the bundledeployment type. Is that really necessary? Looking at corev1 in k8s, it seems `+optional` and `omitempty` is sufficient, unless we really need to differentiate between a null value and an empty map.

Oh, it seems we do: https://github.com/rancher/fleet/blob/9d2019d87d1a276d79a1cb41edfee1f1e141263a/internal/cmd/agent/deployer/deployer.go#L181-L181
That code deletes all annotations if it's an empty map, but not if it's nil. 👍

Reply:
However that would still work with a map, as a map is effectively a pointer.